handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -410,14 +410,12 @@ private static void loadTcNative() throws Exception {
         Set<String> libNames = new LinkedHashSet<String>(4);
         // First, try loading the platform-specific library. Platform-specific
         // libraries will be available if using a tcnative uber jar.
-        libNames.add("netty-tcnative-" + os + '-' + arch);
+        libNames.add("netty_tcnative-" + os + '-' + arch);
         if (LINUX.equalsIgnoreCase(os)) {
             // Fedora SSL lib so naming (libssl.so.10 vs libssl.so.1.0.0)..
-            libNames.add("netty-tcnative-" + os + '-' + arch + "-fedora");
+            libNames.add("netty_tcnative-" + os + '-' + arch + "-fedora");
         }
         // finally the default library.
-        libNames.add("netty-tcnative");
-        // in Java 8, statically compiled JNI code is namespaced
         libNames.add("netty_tcnative");
 
         NativeLibraryLoader.loadFirstAvailable(SSL.class.getClassLoader(),
```

Comment:
it seems like we may not actually use Library class from tnactive to load the native library ... if that is the case we also need to apply the shading prefix here too. And also there is an earlier check in OpenSsl.java to check if its already on the class path [1] which should also take into account the package prefix.

we really need to move tcnative into netty to simplify and reduce duplication 😭 

[1] `Class.forName("io.netty.internal.tcnative.SSL", false, OpenSsl.class.getClassLoader());`